lib/int64.js

```diff
@@ -0,0 +1,385 @@
+    const dSign = d.value[1] >>> 31;
+    const dPositive = new Int64(d);
+    if (dSign) dPositive.value = Int64._conversion(dPositive.value);
+
+    const q = Int64.zero();
+    const r = Int64.zero();
+    for (let i = 63; i >= 0; i--) {
+      r.shiftLeft(1);
+      const valIndex = i >>> 5;
+      const nval = value[valIndex];
+      const ii = i < 32 ? i : i - 32;
+      r.value[0] = (r.value[0] & ~1) | ((nval & (1 << ii)) >>> ii);
+      if (Int64.cmp(r, dPositive) >= 0) {
+        r.sub(dPositive);
+        q.value[valIndex] |= 1 << ii;
+      }
+    }
+    q.value[1] &= INT32_MAX;
+    r.value[1] &= INT32_MAX;
+    if (nSign ^ dSign) q.value = Int64._conversion(q.value);
+    if (nSign) r.value = Int64._conversion(r.value);
+    return [q, r];
+  }
+
+  static div(a, b) {
+    return Int64._division(a, b)[0];
+  }
+
+  static mod(a, b) {
+    return Int64._division(a, b)[1];
+  }
+
+  static mult(a, b) {
+    const zero = Int64.zero();
+    const one = Int64.one();
+    if (Int64.cmp(a, zero) === 0 || Int64.cmp(b, zero) === 0) return zero;
+    if (Int64.cmp(a, one) === 0) return new Int64(b);
+    if (Int64.cmp(b, one) === 0) return new Int64(a);
+
+    const aSign = a.value[1] >>> 31;
+    const bSign = b.value[1] >>> 31;
+    const result = Int64.zero();
+    const aPositive = new Int64(a);
+    let value = new Uint32Array(b.value);
+    if (aSign) aPositive.value = Int64._conversion(aPositive.value);
+    if (bSign) value = Int64._conversion(value);
+
+    for (let i = 0; i < 63; i++) {
+      const valIndex = i >= 32 ? 1 : 0;
+      if (value[valIndex] & 1) {
+        result.add(new Int64(aPositive).shiftLeft(i));
+      }
+      value[valIndex] >>>= 1;
+    }
+    if (aSign ^ bSign) result.value = Int64._conversion(result.value);
+    return result;
+  }
+
+  and(b) {
+    this.value[0] = this.value[0] & b.value[0];
+    this.value[1] = this.value[1] & b.value[1];
+    return this;
+  }
+
+  static and(a, b) {
+    return new Int64(a).and(b);
+  }
+
+  or(b) {
+    this.value[0] = this.value[0] | b.value[0];
+    this.value[1] = this.value[1] | b.value[1];
+    return this;
+  }
+
+  static or(a, b) {
+    return new Int64(a).or(b);
+  }
+
+  not() {
+    this.value[0] = ~this.value[0];
+    this.value[1] = ~this.value[1];
+    return this;
+  }
+
+  static not(a) {
+    return new Int64(a).not();
+  }
+
+  xor(b) {
+    this.value[0] = this.value[0] ^ b.value[0];
+    this.value[1] = this.value[1] ^ b.value[1];
+    return this;
+  }
+
+  static xor(a, b) {
+    return new Int64(a).xor(b);
+  }
+
+  shiftRightLogical(b) {
+    b %= 64;
+    if (b >= 32) {
+      this.value[0] = this.value[1] >>> (b - 32);
+      this.value[1] = 0;
+      return this;
+    }
+    const mask = (1 << b) - 1;
+    const tr = (this.value[1] & mask) << (32 - b);
+    this.value[1] = this.value[1] >>> b;
+    this.value[0] = (this.value[0] >>> b) | tr;
+    return this;
+  }
+
+  shiftRightArithmetic(b) {
+    b %= 63;
```

Comment:
Why do you use 63 and not 64 here?
```suggestion
    b %= 64;
```